components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/listener/JITProvisioningIdentityProviderMgtListener.java

```diff
@@ -0,0 +1,105 @@
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+import org.wso2.carbon.idp.mgt.IdentityProviderManager;
+import org.wso2.carbon.idp.mgt.listener.AbstractIdentityProviderMgtListener;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Listener to perform JIT provisioning related Identity Provider management operations.
+ */
+public class JITProvisioningIdentityProviderMgtListener extends AbstractIdentityProviderMgtListener {
+
+    private static final Log log = LogFactory.getLog(JITProvisioningIdentityProviderMgtListener.class);
+    private static ExecutorService threadPool = Executors.newFixedThreadPool(2);
+
+    @Override
+    public boolean doPostDeleteIdPByResourceId(String resourceId, IdentityProvider identityProvider,
+                                               String tenantDomain) throws IdentityProviderManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("doPostDeleteIdPByResourceId executed for idp: " + resourceId + " of tenant domain: " +
+                    tenantDomain);
+        }
+
+        if (!FrameworkUtils.isEnhancedFeature()) {
+            // JIT provisioning enhanced feature is not enabled. Hence returning.
+            return true;
+        }
+
+        ProvisionedUserDeleteThread provisionedUserDeleteThread =
+                new ProvisionedUserDeleteThread(resourceId, tenantDomain);
+        threadPool.submit(provisionedUserDeleteThread);
+        return true;
+    }
+
+    @Override
+    public boolean doPreDeleteIdP(String idPName, String tenantDomain) throws IdentityProviderManagementException {
+
+        if (!FrameworkUtils.isEnhancedFeature()) {
+            // JIT provisioning enhanced feature is not enabled. Hence returning.
+            return true;
+        }
+
+        String idpId = IdentityProviderManager.getInstance().getIdPByName(idPName, tenantDomain,
+                true).getResourceId();
+        IdentityUtil.threadLocalProperties.get().put(FrameworkConstants.IDP_RESOURCE_ID, idpId);
+        return true;
+    }
+
+    @Override
+    public boolean doPostDeleteIdP(String idPName, String tenantDomain) throws IdentityProviderManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("doPostDeleteIdp executed for idp: " + idPName + " of tenant domain: " + tenantDomain);
+        }
+
+        if (!FrameworkUtils.isEnhancedFeature()) {
+            // JIT provisioning enhanced feature is not enabled. Hence returning.
+            return true;
+        }
```

Comment:
Execute `enhancedFeatureEnabled` check before the debug log or change the log message appropriately.

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3597/commits/6e1589981f43782e7824b22d065010c4ef34b01d